02_bubble/sirigithub/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubbleSort(s []int) []int {
+	n := len(s)
+	arrayToSort := make([]int, n)
+	copy(arrayToSort, s)
```

Comment:
Everything is fine, but just to let you know, you can copy slice using onliner: `b = append(a[:0:0], a...)`  https://github.com/golang/go/wiki/SliceTricks.

Reply:
That is very neat, will keep that in mind. Thank you

Reply:
Just a drive-by comment - I'm in two minds about this pattern for copying a slice. It is quite non-obvious what is going on. At the very least, @n0npax, I think it is best to link to https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F which gives some explanation (but still not really enough for my liking).